calico_upgrade/pkg/calicoupgrade.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/docopt/docopt-go"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/calico/calico_upgrade/pkg/commands"
+	"github.com/projectcalico/libcalico-go/lib/logutils"
+)
+
+func main() {
+	doc := `Usage:
+  calico-upgrade [options] <command> [<args>...]
+
+    dry-run   Perform a dry-run of the data migration. This validates that the
+              v1 formatted data will be successfully converted and that the v3
+              datastore is in the correct state for the data migration. This
+              command outputs a full report of any migrated names, migration
+              errors, or migrated name conflicts. See Description section
+              below for details.
+    start     Start the upgrade process. This does the following:
+              -  performs a dry-run to verify the data will be migrated
+                 successfully
+              -  pauses Calico networking: this prevents new endpoints from
+                 being created while allowing existing endpoints to remain
+                 networked
+              -  migrates the data from v1 to v3 format
+    complete  This resumes Calico networking for the v3.x nodes.
+    abort     This aborts the upgrade process by resuming Calico networking
+              for the v2.x nodes.
+    version   Display the version of calico-upgrade.
+
+Options:
+  -h --help               Show this screen.
+  -l --log-level=<level>  Set the log level (one of panic, fatal, error,
+                          warn, info, debug) [default: panic]
+
+Description:
+  The calico-upgrade command line tool is used to assist with the migration of
+  v1-formatted data to the v3 format used by Calico v3.x.
```

Comment:
This sentence could benefit from a change in voice. May I suggest a more active construction? Perhaps something more like, "Use the calico-upgrade command line tool to migrate manifests from v1 to v3 format." I am not sure, but I think when you say "data" you are referring to the manifests? If so, it would be better to be specific as I have in my suggested rephrasing. I would also recommend going further than I have and specifying just what is at v1. I believe that it is libcalico-go? Or perhaps it is calicoctl?